src/components/_classes/component/Component.js

```diff
@@ -2628,18 +2628,10 @@ export default class Component extends Element {
     this.checkComponentConditions(data, flags, row);
 
     if (flags.noValidate && !flags.validateOnInit) {
-      if (flags.fromSubmission && this.rootPristine && this.pristine && this.error && flags.changed) {
```

Comment:
It seems that this should still be here. From what I can tell, this covers a different use case than an initial validation check.

Reply:
Yes, you are right, it is better to leave this logic. I changed the PR. All tests passed locally.